src/processing.py

```diff
@@ -0,0 +1,40 @@
+# тестовый словарь для проверки
+from datetime import datetime
+
+list_of_dates = [
+    {"id": 41428829, "state": "EXECUTED", "date": "2019-07-03T18:35:29.512364"},
+    {"id": 939719570, "state": "EXECUTED", "date": "2018-06-30T02:08:58.425572"},
+    {"id": 594226727, "state": "CANCELED", "date": "2018-09-12T21:27:25.241689"},
+    {"id": 615064591, "state": "CANCELED", "date": "2018-10-14T08:21:33.419441"},
+]
+
+
+def filter_by_state(list_of_dates: list[dict[str, object]], state: str = "EXECUTED") -> list:
+    """Функция принимает список словарей и возвращает новый список словарей, содержащий только те словари,
+    у которых ключ state
+    соответствует указанному значению: "EXECUTED" или "CANCELED"."""
+
+    filtered = []
+    for entry in list_of_dates:
+        if entry.get("state") == state:
+            filtered.append(entry)
+    return filtered
+
+
+def sort_by_date(list_of_dates: list[dict[str, object]], is_reversed: bool = True) -> list:
```

Comment:
Давай укажем попроще:
```suggestion
def sort_by_date(list_of_dates: list[dict], is_reversed: bool = True) -> list[dict]:
```
Этого пока что достаточно. Главный поинт в том, что мы возвращаем такую же структуру, которую принимаем - это стоит отражать в тайпхинтах

Reply:
Выше в функции тоже самое нужно сделать

Reply:
Если хочешь написать более профессионально - используй [TypedDict](https://peps.python.org/pep-0589/) (просто совет, не призыв к действию)